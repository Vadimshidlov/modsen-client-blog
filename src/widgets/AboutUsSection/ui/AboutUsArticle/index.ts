export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
