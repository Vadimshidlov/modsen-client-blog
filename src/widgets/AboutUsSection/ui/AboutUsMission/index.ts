export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
