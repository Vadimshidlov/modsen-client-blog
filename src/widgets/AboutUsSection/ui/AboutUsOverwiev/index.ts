export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
