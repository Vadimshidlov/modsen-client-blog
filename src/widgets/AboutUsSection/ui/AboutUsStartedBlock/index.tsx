export { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock/AboutUsStartedBlock';
