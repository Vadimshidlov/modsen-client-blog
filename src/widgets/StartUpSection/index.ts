export { StartUpSection } from '@/widgets/StartUpSection/ui/StartUpSection';
