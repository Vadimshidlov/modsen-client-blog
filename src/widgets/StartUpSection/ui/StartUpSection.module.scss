.section__container {
  position: relative;
  max-height: 720px;
  height: 100vh;
  color: $white;

  @media (max-width: 768px) {
    max-height: 620px;
  }

  @media (max-width: 450px) {
    max-height: 550px;
  }

  .content__container {
    padding-top: 128px;
    padding-left: 81px;
    z-index: 1;

    @media (max-width: 768px) {
      padding-top: 90px;
      padding-left: 41px;
      max-height: 520px;
    }

    @media (max-width: 450px) {
      max-height: 420px;
      padding-top: 30px;
      padding-left: 20px;
    }

    .title {
      font-size: $fz56;
      font-weight: $bold;
      line-height: 63px;
      max-width: 803px;
      margin-bottom: 24px;

      @media (max-width: 800px) {
        font-size: $fz40;
        line-height: 43px;
      }
    }

    .subtitle{
      font-size: $fz16;
      font-weight: $semiBold;
      line-height: 20px;
      margin-bottom: 24px;

      span {
        font-weight: $superBold;
      }
    }

    .text__container {
      display: flex;
      flex-direction: column;

      .textTitle {
        font-size: $fz16;
        font-weight: $regular;
        margin-bottom: 16px;

        span {
          color: $yellow;
        }
      }

      .text {
        font-size: $fz16;
        font-weight: $regular;
        max-width: 600px;
        margin-bottom: 48px;
      }
    }
  }
}


