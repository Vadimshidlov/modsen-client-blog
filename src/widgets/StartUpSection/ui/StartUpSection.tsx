import React from 'react';
import Image from 'next/image';
import styles from '@/widgets/StartUpSection/ui/StartUpSection.module.scss';
import startupBackground from '@/widgets/StartUpSection/assets/man-in-black-suit-wearing-eye-glasses-sitting-on-gray-sofa-using-macbook-3772622.jpg';
import { Button } from '@/shared/ui/Button/Button';

export function StartUpSection() {
  return (
    <section className={styles.section__container}>
      <Image
        alt="Mountains"
        src={startupBackground}
        placeholder="blur"
        quality={100}
        fill
        style={{
          objectFit: 'cover',
          zIndex: -1,
        }}
      />
      <div className={styles.content__container}>
        <h3 className={styles.subtitle}>
          POSTED ON&nbsp;
          <span>STARTUP</span>
        </h3>
        <h1 className={styles.title}>Step-by-step guide to choosing great font pairs</h1>
        <div className={styles.text__container}>
          <span className={styles.textTitle}>
            By&nbsp;
            <span>James West</span>
            {' '}
            | May 23, 2022
          </span>
          <span className={styles.text}>
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat
            nulla pariatur. Excepteur sint occaecat cupidatat non proident.
          </span>
          <Button>Read More &gt;</Button>
        </div>
      </div>
    </section>
  );
}
