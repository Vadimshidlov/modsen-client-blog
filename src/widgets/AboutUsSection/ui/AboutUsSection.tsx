import React from 'react';
import { AboutUsTitle } from '@/widgets/AboutUsSection/ui/AboutUsTitle';
import { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev';
import { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission';
import { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam';
import { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock';
import styles from '@/widgets/AboutUsSection/ui/AboutUsSection.module.scss';
import container from '@/shared/styles/container.module.scss';

export function AboutUsSection() {
  return (
    <section className={`${styles.aboutUsSection} ${container.container}`}>
      <AboutUsTitle />
      <AboutUsOverwiev />
      <AboutUsMission />
      <AboutUsTeam />
      <AboutUsStartedBlock />
    </section>
  );
}
