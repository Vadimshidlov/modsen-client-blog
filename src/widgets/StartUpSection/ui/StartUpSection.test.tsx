import React from 'react';
import { render } from '@testing-library/react';
import { StartUpSection } from '@/widgets/StartUpSection/ui/StartUpSection';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('StartUpSection tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<StartUpSection />);
    expect(container).toMatchSnapshot();
  });
});
