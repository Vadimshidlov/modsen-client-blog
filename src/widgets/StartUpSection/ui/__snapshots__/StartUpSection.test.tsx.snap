// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartUpSection tests should matches snapshot 1`] = `
<div>
  <section
    class="section__container"
  >
    <img
      alt="Mountains"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimg.jpg&w=3840&q=100"
      srcset="/_next/image?url=%2Fimg.jpg&w=640&q=100 640w, /_next/image?url=%2Fimg.jpg&w=750&q=100 750w, /_next/image?url=%2Fimg.jpg&w=828&q=100 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=100 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; z-index: -1; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
    />
    <div
      class="content__container"
    >
      <h3
        class="subtitle"
      >
        Mocked titleFirstPart
         
        <span>
          Mocked titleSecondPart
           
        </span>
      </h3>
      <h1
        class="title"
      >
        Step-by-step guide to choosing great font pairs
      </h1>
      <div
        class="text__container"
      >
        <span
          class="textTitle"
        >
          By 
          <span>
            Floyd Miles
          </span>
           |
          May 23, 2024
        </span>
        <span
          class="text"
        >
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
        </span>
        <a
          href="/posts/1"
        >
          <button
            class="button"
            type="button"
          >
            Mocked buttonText
             &gt;
          </button>
        </a>
      </div>
    </div>
  </section>
</div>
`;
